src/Loader/Loader.tsx

```diff
@@ -1,47 +1,56 @@
 import React from "react";
-import "./loader-style.scss";
+import classnames from "classnames";
+import "./loader.scss";
 
-export interface LoaderProps {
-    className?: string;
-    fullscreen?: boolean;
-    id?: string;
+export type LoaderSize = "xs" | "sm" | "md" | "lg";
+export type LoaderType = "spinner" | "square";
+export type LoaderProps = JSX.IntrinsicElements["div"] & {
+    /** Loader size. Supported sizes: `xs`, `sm`, `md`, `lg` */
     size?: LoaderSize;
-    toggle: boolean;
-}
-
-export type LoaderSize = "lg" | "md" | "sm";
-
-export const Loader: React.FunctionComponent<LoaderProps> = React.memo(
-    (props: LoaderProps): React.ReactElement<void> => {
-        const [size, setSize] = React.useState<string>("");
-        const [className, setClassName] = React.useState<string>("seb-loader-wrapper");
+    /** Loader types. Supportes types: `spinner`, `square` */
+    type?: LoaderType;
+    /** Have the loader take over it's parent */
+    cover?: boolean;
+    /** Have the loader take over the screen */
+    fullscreen?: boolean;
+    /** Dims the background the background to indicate UI interactions are blocked */
+    backdrop?: boolean;
+    /** Screen reader text. Default is `Loading...` */
+    srText?: string;
+    /** Show or hide the loader. Default is `true` */
+    toggle?: boolean;
+};
 
-        React.useEffect(() => {
-            let classNameToSet: string = "seb-loader-wrapper";
-            classNameToSet += props.fullscreen !== undefined ? (props.fullscreen ? " fullscreen" : "") : "";
-            classNameToSet += props.className ? ` ${props.className}` : "";
-            setClassName(classNameToSet);
-        }, [props.className, props.fullscreen]);
+export const Loader: React.FC<LoaderProps> = React.memo(({ size = "md", type = "spinner", srText = "Loading...", toggle = true, fullscreen, cover, backdrop, ...props }: LoaderProps) => {
+    const [className, setClassName] = React.useState<string>("");
 
-        React.useEffect(() => {
-            setSize(`loader-${props.size || "md"}`);
-        }, [props.size]);
+    React.useEffect(() => {
+        setClassName(
+            classnames(
+                "rc",
+                "loader",
+                {
+                    "loader-cover": cover,
+                    "loader-fullscreen": fullscreen && !cover,
+                    "loader-backdrop": backdrop,
+                    [`loader-${size}`]: size,
+                    [`loader-${type}`]: type,
+                },
+                props.className
+            )
+        );
+    }, [props.className, cover, fullscreen, size, type]);
```

Comment:
```suggestion
    }, [className, cover, fullscreen, size, type]);
```

Reply:
Same here, I don't need `className` instead I need `props.className`

Reply:
I think it is a bit confusing considering your choice of passing props. Wouldn't `className` still be valid after you remove 

`    const [className, setClassName] = React.useState<string>("");
`
? Can we rename the local className perhaps ?